src/components/encoding-pane/encoding-shelf.tsx

```diff
@@ -28,6 +29,8 @@ export interface EncodingShelfPropsBase extends ActionHandler<ShelfEncodingActio
   id: ShelfId;
 
   fieldDef: ShelfFieldDef;
+
+  schema?: Schema;
```

Comment:
Maybe we should make it required since Field component in encoding shelf does not differentiate wildcard fields from other fields